cpp/ql/src/semmle/code/cpp/ir/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -145,7 +147,10 @@ OutNode getAnOutNode(DataFlowCall call, ReturnKind kind) {
 predicate jumpStep(Node n1, Node n2) { none() }
 
 private newtype TContent =
-  TFieldContent(Field f) or
+  TFieldContent(IntValue startBitOffset, IntValue endBitOffset) {
+    getDefInterval(_, startBitOffset, endBitOffset) or
+    getUseInterval(_, startBitOffset, endBitOffset)
```

Comment:
Could this be an `and` rather than an `or`? Why track flow through fields that are never written to or never read from?

(I think the shared data-flow library may already do this optimisation, but I'm not sure how late it happens.)

Reply:
Good point. I think it's safe to make this a conjunction. I'll try it out locally and check if it makes a performance difference on Jenkins.

Reply:
I decided to limit the values in this type differently in 9de570b300dd0cdbe336549d79a4b7b1f1197c99 (instead they're not limited to only those entries that correspond to a `Field`).